### ISSUE TITLE
Error:  Invalid cross-device link when using named volumes for data directory

### ISSUE THREAD

**Issue by getaju69** [tianon/docker-postgres-upgrade#41]
I am trying to upgrade my postgres 10 to 14 (on my windows machine)

When I do this, I get the error:

```
could not create hard link between old and new data directories: Invalid cross-device link
In link mode the old and new data directories must be on the same file system.
Failure, exiting
```

This is the command I used:

```
docker run --rm \
  -v postgresql-data-volume:/var/lib/postgresql/10/data \
  -v postgresql-data-volume-14:/var/lib/postgresql/14/data \
  tianon/postgres-upgrade:10-to-14 --link
```

It works, when I do not use the link option. I would need the link option to do it (as I need the same thing to be also done on production)

I would think it is due to the named volumes I use (just like how it is configured on production)

```
postgresql-data-volume
postgresql-data-volume-14:
```

Is it possible to do the in place upgrade using different named volumes, or should I be using a local directory for the same (it seems I cannot do this on windows - using named directory)

(running the test setup as per the readme file worked without any issues)

In short, does it also support a **--link upgrade for named volumes** ?
